src/Libraries/Nop.Data/Migrations/UpgradeTo470/AddIndexesMigration.cs

```diff
@@ -0,0 +1,104 @@
+				.OnColumn(nameof(ForumPost.CustomerId)).Ascending()
+				.WithOptions().NonClustered();
+
+			IfDatabase(databaseType).Create.Index("IX_CustomerPassword_CustomerId").OnTable(nameof(CustomerPassword))
+				.OnColumn(nameof(CustomerPassword.CustomerId)).Ascending()
+				.WithOptions().NonClustered();
+
+			IfDatabase(databaseType).Create.Index("IX_ExternalAuthenticationRecord_CustomerId").OnTable(nameof(ExternalAuthenticationRecord))
+				.OnColumn(nameof(ExternalAuthenticationRecord.CustomerId)).Ascending()
+				.WithOptions().NonClustered();
+
+			IfDatabase(databaseType).Create.Index("IX_PrivateMessage_From_CustomerId").OnTable(nameof(PrivateMessage))
+				.OnColumn(nameof(PrivateMessage.FromCustomerId)).Ascending()
+				.WithOptions().NonClustered();
+
+			IfDatabase(databaseType).Create.Index("IX_PrivateMessage_To_CustomerId").OnTable(nameof(PrivateMessage))
+				.OnColumn(nameof(PrivateMessage.ToCustomerId)).Ascending()
+				.WithOptions().NonClustered();
+
+			IfDatabase(databaseType).Create.Index("IX_ForumSubscription_CustomerId").OnTable(nameof(ForumSubscription))
+				.OnColumn(nameof(ForumSubscription.CustomerId)).Ascending()
+				.WithOptions().NonClustered();
+
+			IfDatabase(databaseType).Create.Index("IX_PollVotingRecord_CustomerId").OnTable(nameof(PollVotingRecord))
+				.OnColumn(nameof(PollVotingRecord.CustomerId)).Ascending()
+				.WithOptions().NonClustered();
+
+			IfDatabase(databaseType).Create.Index("IX_Order_CustomerId").OnTable(nameof(Order))
+				.OnColumn(nameof(Order.CustomerId)).Ascending()
+				.WithOptions().NonClustered();
+
+			IfDatabase(databaseType).Create.Index("IX_Log_CustomerId").OnTable(nameof(Log))
+				.OnColumn(nameof(Log.CustomerId)).Ascending()
+				.WithOptions().NonClustered();
+
+			IfDatabase(databaseType).Create.Index("IX_BlogComment_CustomerId").OnTable(nameof(BlogComment))
+				.OnColumn(nameof(BlogComment.CustomerId)).Ascending()
+				.WithOptions().NonClustered();
+
+			IfDatabase(databaseType).Create.Index("IX_BackInStockSubscription_CustomerId").OnTable(nameof(BackInStockSubscription))
+				.OnColumn(nameof(BackInStockSubscription.CustomerId)).Ascending()
+				.WithOptions().NonClustered();
+
+			IfDatabase(databaseType).Create.Index("IX_ReturnRequest_CustomerId").OnTable(nameof(ReturnRequest))
+				.OnColumn(nameof(ReturnRequest.CustomerId)).Ascending()
+				.WithOptions().NonClustered();
+
+			IfDatabase(databaseType).Create.Index("IX_RewardPointsHistory_CustomerId").OnTable(nameof(RewardPointsHistory))
+				.OnColumn(nameof(RewardPointsHistory.CustomerId)).Ascending()
+				.WithOptions().NonClustered();
+
+			IfDatabase(databaseType).Create.Index("IX_NewsComment_CustomerId").OnTable(nameof(NewsComment))
+				.OnColumn(nameof(NewsComment.CustomerId)).Ascending()
+				.WithOptions().NonClustered();
+
+			IfDatabase(databaseType).Create.Index("IX_CustomerAddressMapping_CustomerId").OnTable(nameof(CustomerAddressMapping))
```

Comment:
As with CustomerCustomerRoleMapping, you must use NameCompatibilityManager for both the table and the CustomerId field

Reply:
NameCompatibilityManager